server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -182,6 +184,21 @@ public int setCurrentVersion(long secretId, long secretContentId) {
     return ImmutableList.copyOf(r);
   }
 
+  public ImmutableList<SecretSeries> getSecretSeriesBatched(int idx, int num) {
```

Comment:
Good point (I'm not certain how MySQL handles negative values, but it should be avoided)